test/node-unit/mocha.spec.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const path = require('path');
+const Mocha = require('../../lib/mocha');
+const utils = require('../../lib/utils');
+
+describe('Mocha', function() {
+  const opts = {reporter: utils.noop}; // no output
+  const testFiles = [
+    __filename,
+    path.join(__dirname, 'cli', 'config.spec.js'),
+    path.join(__dirname, 'cli', 'run.spec.js')
+  ];
+  const resolvedTestFiles = testFiles.map(require.resolve);
+
+  describe('#addFile', function() {
+    it('should add the given file to the files array', function() {
+      const mocha = new Mocha(opts);
+      mocha.addFile(__filename);
+      expect(mocha.files, 'to have length', 1).and('to contain', __filename);
+    });
+
+    it('should be chainable', function() {
+      const mocha = new Mocha(opts);
+      expect(mocha.addFile(__filename), 'to be', mocha);
+    });
+  });
+
+  describe('#loadFiles', function() {
+    it('should load all files from the files array', function() {
+      const mocha = new Mocha(opts);
+
+      testFiles.forEach(mocha.addFile, mocha);
+      mocha.loadFiles();
+      expect(require.cache, 'to have keys', resolvedTestFiles);
+    });
+
+    it('should execute the optional callback if given', function() {
+      const mocha = new Mocha(opts);
+      expect(cb => {
```

Comment:
I don't think I've ever used "to call the callback" before (as I usually delegate this stuff to sinon), so it's cool to learn about this.  but... would maybe be better to be more specific about whether the callback is called with an error or not? 
 "to call the callback without error" looks like the syntax.

Reply:
It's a no-arg callback (Function.length === 0), so not really relevant in this case.

```js
Mocha.prototype.loadFiles = function(fn) {
  var self = this;
  var suite = this.suite;
  this.files.forEach(function(file) {
    file = path.resolve(file);
    suite.emit(EVENT_FILE_PRE_REQUIRE, global, file, self);
    suite.emit(EVENT_FILE_REQUIRE, require(file), file, self);
    suite.emit(EVENT_FILE_POST_REQUIRE, global, file, self);
  });
  fn && fn();
};
```